File: ejc_sistema/models/participant.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class Participant:
    id: Optional[int]
    name: str
    common_name: Optional[str] = None
    birth_date: Optional[str] = None  # ISO YYYY-MM-DD
    instagram: Optional[str] = None
    address: Optional[str] = None
    neighborhood: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    sacraments: Optional[str] = None
    church_movement: Optional[str] = None
    father_name: Optional[str] = None
    father_contact: Optional[str] = None
    mother_name: Optional[str] = None
    mother_contact: Optional[str] = None
    ecc_participant: Optional[bool] = None
    ecc_info: Optional[str] = None
    has_restrictions: Optional[bool] = None
    restrictions_info: Optional[str] = None
    observations: Optional[str] = None
    photo_path: Optional[str] = None

File: src/types/participant.ts
export interface Participant {
  id: number
  name: string
  common_name?: string | null
  birth_date?: string | null
  instagram?: string | null
  address?: string | null
  neighborhood?: string | null
  email?: string | null
  phone?: string | null
  sacraments?: string | null
  church_movement?: string | null
  church_movement_info?: string | null
  father_name?: string | null
  father_contact?: string | null
  mother_name?: string | null
  mother_contact?: string | null
  ecc_participant?: boolean | null
  ecc_info?: string | null
  has_restrictions?: boolean | null
  restrictions_info?: string | null
  observations?: string | null
  photo_path?: string | null
}

export interface ParticipantCreate {
  name: string
  common_name?: string | null
  birth_date?: string | null
  instagram?: string | null
  address?: string | null
  neighborhood?: string | null
  email?: string | null
  phone?: string | null
  sacraments?: string | null
  church_movement?: string | null
  church_movement_info?: string | null
  father_name?: string | null
  father_contact?: string | null
  mother_name?: string | null
  mother_contact?: string | null
  ecc_participant?: boolean | null
  ecc_info?: string | null
  has_restrictions?: boolean | null
  restrictions_info?: string | null
  observations?: string | null
  photo_path?: string | null
}

export interface ParticipantUpdate extends Partial<ParticipantCreate> {}

export interface SacramentStatus {
  sacrament: 'Batismo' | 'Primeira Eucaristia' | 'Crisma'
  status: 'Concluído' | 'Não Concluído' | 'Em Processo' | 'Não Informado'
}
